lesson4/index.md

```diff
@@ -0,0 +1,239 @@
+        Scanner input = new Scanner(System.in);
+
+        int numberToGuess = 19;
+        boolean numberGuessed = false;
+        int neededTries = 0;
+        
+        do {
+            neededTries++;
+            System.out.println("Please guess a number between 0 and 20");
+
+            int guessedNumber = input.nextInt();
+
+            if (guessedNumber == numberToGuess) {
+                numberGuessed = true;
+            }          
+        } while (!numberGuessed);
+
+        System.out.println("Congratulations! You needed " + neededTries + " tries to guess the given number!");
+    }
+}
+```
+
+## for each loops
+For each loops are like normal for loops. However, a common case is to iterate over a whole list/a whole array.
+
+Structure:
+```java
+for (type var : array) { 
+    <CODE TO EXECUTE>
+}
+```
+This will iterate of the whole array and assign each value once to the variable var.
+
+Example:
+```java
+for (int number : numbers) {
+    System.out.println(number);
+}
+```
+The above prints the same results as the following for loop below:
+```java
+for (int i=0; i<numbers.length; i++) {
+    System.out.println(numbers[i]);
+}
+```
+
+A for each loop is quite common when you handle lists/arrays.
+
+## Exercises
+### Exercise 1: Multiplication Table
+- Read a one number with Scanner
+- Print the multiplication table of the number from 1 till 10
+
+Example:
+```
+Input: 2
+Output:
+2 x 1 = 2
+2 x 2 = 4
+2 x 3 = 6
+2 x 4 = 8
+2 x 5 = 10
+2 x 6 = 12
+2 x 7 = 14
+2 x 8 = 16
+2 x 9 = 18
+2 x 10 = 20
+```
+
+### Exercise 2: Dividable numbers
+- Read two numbers with Scanner
+- Print each number from 1 to the first entered number that is dividable without remainder by the second number (e.g. 6 is dividable by 3 without remainder as 6 / 3 is 2 with remainder 0, but 6 is not dividable by 4 without remainder as 6 / 4 is 1 with remainder 2)
+
+Example:
+```
+First: 16
+Second: 5
+Output:
+5 is dividable by 5
+10 is dividable by 5
+15 is dividable by 5
+```
+### Exercise 3: Bank Account
+- Your program should start with a variable holding the current money on a user’s bank account (start with e.g. 100)
+- The user should now enter an amount and he should enter if he wants to deposit to his account or if he wants to withdraw (e.g. 1 for deposit, 2 for withdrawal)
+- The variable holding money on the account should be updated accordingly and the current money should be printed.
+The program should run and let the user interact until the money on his account is zero or less.
+
+Example:
+```
+Current money on account: 100
+Enter amount:
+30
+Enter 1 for deposit, 2 for withdrawal
+2
+Current money on account: 70
+Enter amount:
+60
+Enter 1 for deposit, 2 for withdrawal
+2
+Current money on account: 10
+Enter amount:
+20
+Enter 1 for deposit, 2 for withdrawal
+1
+Current money on account: 30
+Enter amount:
+30
+Enter 1 for deposit, 2 for withdrawal
+2
+You have no money left, program stopped ...
+```
+### Exercise 4: Create a pattern
```

Comment:
This one was their homework from lesson 2 :)

Reply:
Hahahaha. Ok, gonna change it